test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart

```diff
@@ -0,0 +1,472 @@
+    mailboxDashboardController.accountId.value = testAccountId;
+  });
+
+  tearDown(Get.deleteAll);
+
+  test('WHEN user search email by keyword, '
+    'THEN user filter search result, '
+    'THEN user tap on mail box, '
+    'SHOULD reset all search and filter options, '
+    'THEN query get all email with default options',
+  () async {
+    // arrange
+    when(context.owner).thenReturn(BuildOwner(focusManager: FocusManager()));
+    
+    // expect query in search controller update as expected
+    mailboxDashboardController.searchEmail(context, queryString: queryString);
+    expect(searchController.searchEmailFilter.value.text, SearchQuery(queryString));
+    
+    // expect sort in search controller update as expected
+    mailboxDashboardController.selectSortOrderQuickSearchFilter(
+      context,
+      EmailSortOrderType.oldest);
+    expect(searchController.sortOrderFiltered.value, EmailSortOrderType.oldest);
+
+    // expect filter in search controller update as expected
+    mailboxDashboardController.selectQuickSearchFilterAction(QuickSearchFilter.hasAttachment);
+    expect(searchController.searchEmailFilter.value.hasAttachment, true);
+    mailboxDashboardController.selectQuickSearchFilterAction(QuickSearchFilter.last7Days);
+    mailboxDashboardController.selectReceiveTimeQuickSearchFilter(context, EmailReceiveTimeType.last30Days);
+    expect(searchController.searchEmailFilter.value.emailReceiveTimeType, EmailReceiveTimeType.last30Days);
+
+    // expect mailbox dashboard controller calls GetEmailsInMailboxInteractor
+    // when [selectedMailbox] is changed and triggered obx listener in thread controller
+    mailboxController.openMailbox(context, PresentationMailbox(testMailboxId));
+    await untilCalled(getEmailsInMailboxInteractor.execute(
+      any, any,
+      limit: anyNamed('limit'),
+      sort: anyNamed('sort'),
+      emailFilter: anyNamed('emailFilter'),
+      propertiesCreated: anyNamed('propertiesCreated'),
+      propertiesUpdated: anyNamed('propertiesUpdated')));
+    expect(searchController.sortOrderFiltered.value, EmailSortOrderType.mostRecent);
+    expect(searchController.searchEmailFilter.value, SearchEmailFilter.initial());
+    verify(getEmailsInMailboxInteractor.execute(
+      testSession, testAccountId,
+      limit: ThreadConstants.defaultLimit,
+      sort: EmailSortOrderType.mostRecent.getSortOrder().toNullable(),
+      emailFilter: EmailFilter(
+        filter: EmailFilterCondition(inMailbox: testMailboxId),
+        filterOption: FilterMessageOption.all,
+        mailboxId: testMailboxId),
+      propertiesCreated: ThreadConstants.propertiesDefault,
+      propertiesUpdated: ThreadConstants.propertiesUpdatedDefault));
+  });
+
+  test('WHEN user use advanced search, '
+    'THEN user tap on mail box, '
+    'SHOULD reset all advanced search and filter options, '
+    'THEN query get all email with default options',
+  () async {
```

Comment:
how about test case: change sort option in advanced search also?

Reply:
The test description is misleading. The second test was covering all cases in advanced search/sort/filter. Will update the test description.

Reply:
Done